Atomic/syscontainers.py

```diff
@@ -2779,6 +2785,89 @@ def _correct_dir_permissions_for_user(path):
                     s = os.stat(fullpath)
                     os.chmod(fullpath, s.st_mode | 0o600)
 
+    def _configure_userns(self, config):
+        """
+        Configure the userns for the specified OCI configuration.
+        """
+        config['linux']['namespaces'] = [i for i in config['linux']['namespaces'] if i['type'] != 'network']
+
+        namespaces = set([i['type'] for i in config['linux']['namespaces']])
+
+        # force the 'user' namespace
+        if 'user' not in namespaces:
+            config['linux']['namespaces'].append({'type' : 'user'})
+            namespaces.add('user')
+
+        all_namespaces = set(['pid', 'uts', 'network', 'mount', 'ipc', 'user'])
+
+        drop_sys = len(all_namespaces - namespaces) > 0
+
+        new_mounts = []
+        for i in config['mounts']:
+            if drop_sys and i['destination'].startswith("/sys"):
+                continue
+            if 'options' not in i:
+                i['options'] = []
+
+            if not i['destination'].startswith('/dev'):
+                if 'nodev' not in i['options']:
+                    i['options'].append('nodev')
+                if 'nosuid' not in i['options']:
+                    i['options'].append('nosuid')
+            new_mounts.append(i)
+
+        if drop_sys:
+            sys_mount = {
```

Comment:
Minor and question: below is the default generated in runc spec, would the type of mount matter this case?
 ```
"destination": "/sys",
                        "type": "none",
                        "source": "/sys",
                        "options": [
                                "rbind",
                                "nosuid",
                                "noexec",
                                "nodev",
                                "ro"
                        ]
```

Reply:
yes, if not all the namespaces are used then mounting sysfs fails.  This is the check I've added and in that case we bind mount it from the host.

Reply:
Makes sense, but my only concern is that will overwrite the default (the one without network in its namespaces) `/sys` if there is one.

For instance, if we input default configuration generated by `runc spec --rootless`, the mount type will be overwritten to "bind", would that matter?

Reply:
The default configuration given by `runc spec --rootless` is a bind mount as the one we are adding here

Reply:
wot, ahh, I guess it must be version difference then, my runc version is `runc.x86_64 2:1.0.0-19.rc5.git4bb1fe4.fc27`, and that gives me none for mounting type of `/sys`. SGTM then, thanks for the explanation. :)

Reply:
the type=none fallbacks to a bind mount in runc :-)  but the correct type is `bind`

Reply:
Can you do --net=none?

Reply:
yes we could do that, that means have a network namespace but not initialize it.  I think that would not be very useful by default, and initialize the network for the rootless case would still require a `setuid` program.  So I preferred to use the host network namespace by default and if the user wishes the config.json file can be edited as we don't offer all the options from the atomic CLI to do it.